File: 2408-number-of-people-aware-of-a-secret/number-of-people-aware-of-a-secret.py
class Solution:
    def peopleAwareOfSecret(self, n: int, delay: int, forget: int) -> int:
        dp = [0 for _ in range(n)]
        dp[0] = 1

        knowns = 0

        for day in range(1, n):
            dp[day] = knowns + dp[day - delay] - dp[day - forget]
            knowns = dp[day]

        return sum(dp[n - forget:]) % 1_000_000_007

File: 2408-number-of-people-aware-of-a-secret/number-of-people-aware-of-a-secret.js
const MOD = 1_000_000_007

/**
 * @param {number} n
 * @param {number} delay
 * @param {number} forget
 * @return {number}
 */
var peopleAwareOfSecret = function(n, delay, forget) {
    const dp = new Array(n).fill(0);
    dp[0] = 1;

    let known = 0;

    for (let day = 1; day < n; day ++) {
        dp[day] = (known + dp.at(day - delay) - dp.at(day - forget) + MOD) % MOD;
        known = dp[day];
    }
    return dp
            .slice(n - forget, n)
            .reduce((acc, cur) => (acc + cur) % MOD, 0)
};